jetty-server/src/main/java/org/eclipse/jetty/server/session/AbstractSessionCache.java

```diff
@@ -366,10 +368,15 @@ protected Session getAndEnter(String id, boolean enter) throws Exception
             catch (Exception e)
             {
                 LOG.warn("Error loading session {}", id, e);
+                exception.set(e);
                 return null;
             }
         });
 
+        Exception ex = exception.get();
+        if (ex != null)
+            throw ex;
```

Comment:
Since now you are rethrowing the exception, you don't want to log it at warn level at 370 - otherwise it will be printed out twice.